apps/studio/src/lib/db/clients/postgresql.ts

```diff
@@ -1189,7 +1189,9 @@ async function updateValues(cli: any, rawUpdates: TableUpdate[]): Promise<TableU
   const updates = rawUpdates.map((update) => {
     const result = { ...update}
     if (update.columnType?.startsWith('_')) {
-      result.value = JSON.parse(update.value)
+      const extracted: string | undefined = update.value.match(/\{(.*)\}/)?.[1]
```

Comment:
Hmm, this seems pretty brittle. This would match strings with `{` legitimately in them.

How about we use postgres-array? It's a package on NPM - https://github.com/bendrucker/postgres-array

That way we don't need to own the parsing. Generally if I see a regex with database values that makes my spidey-sense go off a little bit because I did this a lot in the beginning and caused a lot of bugs :-D

Reply:
lol, now I see that you mentioned it.

Reply:
Since we are changing the way we write array value from `[1,2,3]` to `{1,2,3}`, should we supply them as they are, or do we still need to parse them into arrays?

Reply:
Sorry, do you mean should we change how they are displayed to the user in results/table?

Reply:
I feel like it's more natural to write the original syntax `{1,2,3}` in the table view rather than `[1,2,3]`. We can also avoid issues with parsing since we probably don't need it, but it's up to you.

Reply:
I had a little issue with the library you mentioned. It gave me some bad result when I typed: 
- parse(`{{1,2}, {3,4}}`) => `[ [ '1', '2' ], [ '3', '4' ], ' ' ]` ❌ 
- parse(`{{1,2},{3,4}}` )=> `[ [ '1', '2' ], [ '3', '4' ] ]` ✅

Reply:
@not-night-but yes, and also to send them to the client. Right now we send array to the client.

Sorry, I didn't see your reply @not-night-but, github didn't show it after I refresh the page 😅

Reply:
Oh no problem! With the amount of hours between us we're bound to have very scattered conversations 😂. I agree with you, we should show arrays in the postgres syntax. It means those values can be directly pasted into a query, and may save us some conversions.

With that other thing, that is incredibly confusing, but it isn't necessarily wrong? I'm not super familiar with postgres arrays, but we could probably just filter the result for any empty values, no?

Reply:
That could do 🤔 . But it gets even messier if I type `{ "a" , "b" }`, and it gives me this result `[ ' a', ' ', ' b', ' ' ]`.

Reply:
Hmmm yeah that's annoying. Seems to be an issue with the parser itself. Probably gets even worse with multidimensional arrays as well 😡. 

From the examples you've given it seems that the parser is expecting no spaces at all... What happens if you input something like `{"hello there","hi"}`?

Reply:
Yeah I think it expects no spaces. That's just `[ 'hello there', 'hi' ]`.

Reply:
Are you inputting these values manually? Or are they coming from a Postgres database. I wonder if postgres removes the extraneous spaces when it stores and that's why the lib doesn't parse the spaces properly. If that's not the case then we're probably blocked on whether we want to just file a PR on the original repo, fork it and make the necessary changes ourselves, or just implement our own parser 🤷

Reply:
Yes, it's manual. I mean, when you try to add rows of records, or applying some changes in the table view, some users probably would use extra spaces which is reasonable. And that's ok in postgres. I'll add some tests related to this to consolidate the case.

Reply:
I just mean if you create a table with an array column, insert something like `{ "1" , "2" , "3" }`, what gets returned when selected? Does postgres do any sanitization before storing the record?

Reply:
I think postgres remove the spaces internally, if I run `insert into ... ('{ "1" , "2" }')` and select it , it will return `{"1","2"}`.

Reply:
So at least that explains why this was overlooked in the library. I'm sure @rathboma will have some ideas when he wakes up lol

Reply:
Hi Azmy,
I think for now, if it's easier to keep it as `{}`, then lets do that.